tests/integration/transactions/test_batch.py

```diff
@@ -0,0 +1,51 @@
+from tests.integration.integration_test_case import IntegrationTestCase
+from tests.integration.it_utils import (
+    sign_and_reliable_submission_async,
+    test_async_and_sync,
+)
+from tests.integration.reusable_values import DESTINATION, WALLET
+from xrpl.asyncio.transaction import autofill
+from xrpl.models import Batch, BatchFlag, Payment
+from xrpl.models.response import ResponseStatus
+from xrpl.transaction.batch_signers import sign_multiaccount_batch
+
+
+class TestBatch(IntegrationTestCase):
+    @test_async_and_sync(globals())
+    async def test_basic_functionality(self, client):
+        payment = Payment(
+            account=WALLET.address,
+            amount="1",
+            destination=DESTINATION.address,
+        )
+        batch = Batch(
+            account=WALLET.address,
+            flags=BatchFlag.TF_ALL_OR_NOTHING,
+            raw_transactions=[payment, payment],
+        )
+        response = await sign_and_reliable_submission_async(batch, WALLET, client)
+        self.assertEqual(response.status, ResponseStatus.SUCCESS)
+        self.assertEqual(response.result["engine_result"], "tesSUCCESS")
+
+    @test_async_and_sync(globals(), ["xrpl.transaction.autofill"])
+    async def test_multisign(self, client):
+        payment = Payment(
+            account=WALLET.address,
+            amount="1",
+            destination=DESTINATION.address,
+        )
+        payment2 = Payment(
+            account=DESTINATION.address,
+            amount="1",
+            destination=WALLET.address,
+        )
+        batch = Batch(
+            account=WALLET.address,
+            flags=BatchFlag.TF_ALL_OR_NOTHING,
+            raw_transactions=[payment, payment2],
+        )
+        autofilled = await autofill(batch, client, 2)
```

Comment:
Do we expect the users to accurately fill-in the value for `signers_count` in this `autofill` method?

If you execute this test with `autofilled = await autofill(batch, client)` -- I have not specified the number of `signers_count`, then the test fails (with `telINSUF_FEE_P` error).

This is because both the `_check_fee` and the `autofill` methods incorrectly assume that the `signers_count == 0`.

Whose responsibility is it to accurately fill the `signers_count` value? 
- If we ask the users to do it, it degrades the user-experience. Users need to be careful in computing the total number of signers, especially in cases where the multi-signing accounts also have `SignerLists` set up for them.

- Do we have any utility methods which compute the `signer_count` automatically?

Reply:
Unfortunately, there's no way to calculate it automatically - it has to be done by the user. `autofill` is executed before any of the signatures are there, and we need the signatures to know how many signers are going to be there. Theoretically users could fill in the rest of the e.g. `Signer` fields (`Account`/`PublicKey`) but in practice that's really not worth the effort just for the purpose of `autofill`, it's much easier and more user-friendly to fill it in by hand.